src/App.tsx

```diff
@@ -1,14 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+// import React, { useCallback, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState('');
+
+  let visibleTodos = [...todos];
+
+  if (query) {
+    const preparedQuery = query.trim().toLowerCase();
+
+    visibleTodos = visibleTodos.filter(todo => todo.title.toLowerCase().includes(preparedQuery));
+  }
+
+  switch (filterBy) {
+    case 'active':
+      visibleTodos = visibleTodos.filter(todo => !todo.completed);
+      break;
+    case 'completed':
+      visibleTodos = visibleTodos.filter(todo => todo.completed);
+      break;
+    case 'all':
+    default:
+      visibleTodos = [...visibleTodos];
+      break;
+  }
+
+  const fetchData = async () => {
+    const data = await getTodos();
```

Comment:
I think. will be better to add handling errors here, if server won't be work properly